### PR TITLE
Add BLE dependency on MQTT for FreeRTOS console

### PR DESCRIPTION
**Pull request by aggarw13**
Equivalent of #2626

### CODE REVIEW

**Review by ravibhagavandas — APPROVED** [review]

File: libraries/core_mqtt_demo_dependencies.cmake [hunk 1]
```diff
@@ -99,4 +99,15 @@ if(TARGET AFR::wifi::mcu_port)
         PUBLIC
             AFR::wifi
     )
+endif()
+
+# Add dependency on BLE module so that the BLE library is auto-included
+# when selecting core MQTT library on FreeRTOS console for boards that
```
Comment (gkwicker) [inline libraries/core_mqtt_demo_dependencies.cmake:105]: Is this just for the console use case?  If someone just clones this repo and builds this demo, would they need lines 107-112?  If so I'd reword the comment on lines 104-106:

```sh
# Add dependency on BLE module for boards that support BLE.
```

**Review by gkwicker — CHANGES_REQUESTED** [review]
See comment

File: libraries/core_mqtt_demo_dependencies.cmake (hunk 1 above)
Comment (aggarw13) [reply libraries/core_mqtt_demo_dependencies.cmake:105]: Nope, these changes do not affect the demo configuration as the demos themselves have the `AFR::ble` dependency specified in their CMake configuration. 
These changes are purely for the FreeRTOS console functionality.

**Review by gkwicker — APPROVED** [review]
